Doc/library/typing.rst

```diff
@@ -585,6 +587,22 @@ These can be used as types in annotations and do not support ``[]``.
    .. versionadded:: 3.5.4
    .. versionadded:: 3.6.2
 
+.. data:: Self
+
+   Special annotation to represent the current enclosed class.
+   For Example::
+
+      from typing import Self
+
+      class ReturnsSelf:
```

Comment:
Class itself does not return `Self`, one of its methods does.

Reply:
```py
class Foo:
    def returns_self(self) -> Self:
```
Would this be better?